src/modules/Game.class.js

```diff
@@ -50,17 +158,97 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.#status;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.#addNumber();
+    this.#addNumber();
+    this.#status = 'playing';
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
+    this.#applyInitialState();
+    this.#score = 0;
+    this.#status = 'idle';
+  }
+
+  #applyInitialState() {
+    this.#state = this.#initialState.map((row) => [...row]);
+  }
+  /**
+   * Checks if any move is possible
+   * @returns {boolean}
+   */
+  #isMovePossible() {
+    const gameCopy = new Game(this.getState(), true);
+
+    return (
+      gameCopy.moveLeft() ||
+      gameCopy.moveRight() ||
+      gameCopy.moveUp() ||
+      gameCopy.moveDown()
+    );
```

Comment:
The '#isMovePossible' method creates a new Game instance to check for possible moves. This might be inefficient as it involves creating multiple instances and performing moves on them. Consider optimizing this method to avoid unnecessary object creation.